LockExample/FileService.cs

```diff
@@ -0,0 +1,77 @@
+﻿using System.Text;
+namespace LockExample
+{
+    internal static class FileService
+    {
+        private const string ResultFileName = "result.txt";
+        private const string FolderName = "LockStatementApp";
+        private static readonly object fileLock = new();
+
+        public static string GetPathDirectory(string dirFilename)
+        {
+            var dirApp = Environment.GetFolderPath(Environment.SpecialFolder.LocalApplicationData);
+            var locationFile = Path.Combine(dirApp, FolderName, dirFilename);
+            var dir = Path.GetDirectoryName(locationFile);
+            if (!Directory.Exists(dir))
+            {
+                if (dir is not null)
+                {
+                    _ = Directory.CreateDirectory(dir);
+                }
+            }
+            return dir ?? "";
+        }
+        public static void UpdateFile(string fileName)
+        {
+            var path = GetPathDirectory(fileName);
+            var fullPath = Path.Combine(path, fileName);
+            var pathToFileResult = Path.Combine(path, ResultFileName);
+            string? textResultRead;
+            lock (fileLock)
+            {
+                if (!File.Exists(fullPath))
+                {
+                    try
+                    {
+                        var textWrite = $"This is some text in the {fileName} file.";
+                        CreateAndWriteToFile(fullPath, textWrite);
+                        textResultRead = ReadFromFile(fullPath);
+                        EnsureFile(pathToFileResult, textResultRead);
```

Comment:
Consider putting the action that can interfere with each in the lock. 
Now too many actions are in lock.